File: 1137. N-th Tribonacci Number/tribonacci.go
func tribonacci(n int) int {
	dp := [3]int{0, 1, 1}
	sum := 2
	prev := 0
	for i := 3; i <= n; i++ {
		prev = dp[(i-3)%3]
		dp[i%3] = sum
		sum = sum*2 - prev
	}
	return dp[n%3]
}

File: 1137. N-th Tribonacci Number/tribonacci.js
/**
 * @param {number} n
 * @return {number}
 */
var tribonacci = function(n) {
    let dp = [0, 1, 1];
    let cursum = 2;
    for (let i = 3; i <= n; i++) {
        let prev = dp[(i-3)%3];
        dp[i%3] = cursum;
        cursum = 2*cursum-prev;
    }
    return dp[n%3];    
};